app/controllers/expenses_controller.rb

```diff
@@ -0,0 +1,81 @@
+class ExpensesController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_expense, only: %i[show edit update destroy]
+
+  # GET /expenses or /expenses.json
+  def index
+    @group = current_user.groups.find(params[:group_id])
+    @expenses = @group.expenses..order(created_at: :desc)
+  end
+
+  # GET /expenses/1 or /expenses/1.json
+  def show; end
+
+  # GET /expenses/new
+  def new
+    @group = current_user.groups.find_by(id: params[:group_id])
+    @expense = @group.expenses.build
+  end
+
+  # GET /expenses/1/edit
+  def edit
+    @group = current_user.groups.find_by(id: params[:group_id])
+    @expense = @group.expenses.find(params[:id])
+  end
+
+  # POST /expenses or /expenses.json
+  def create
+    @group = current_user.groups.find_by(id: params[:group_id])
```

Comment:
The use of ` @group = current_user.groups.find(params[:group_id])` across your controller violates the DRY principle. You can declare it once at instance level `def new` and use everywhere in you controller

- [ ] Kindly refactor your controller

Reply:
- [ ] Hello @AshakaE thank you for your review, kindly explain further what exactly I should fix on the video.

- [ ] also for Heroku, it is no longer offering free membership and my peers didn't manage to deploy as well, if you are available to assist can we have a zoom call?

Reply:
With respect to your video presentation, what I meant is `The video contains a highlight about a piece of the code the student finds interesting or is proud of making.`

Reply:
Hello @AshakaE  thank you for pointing out refactoring but this is not duplication , I tried to make the refactoring but it gives me an error .

https://www.loom.com/share/f0c87874410d4e3cb5bc56f4e1212eca